### PR TITLE
Implement SelectNodeId Component

### PR DESCRIPTION
**Pull request by MohamedElmdary**

### CODE REVIEW

File: src/components/select_node_id.vue [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  </section>
+</template>
+
+<script lang="ts" setup>
+import { ref, watch, onMounted, computed } from 'vue'
+import { useProfileManager } from '../stores/profile_manager'
+import { gridProxyClient } from '../clients'
+import { getGrid } from '../utils/grid'
+import type { NodeInfo } from 'grid3_client'
+import { NodeStatus } from 'tf_gridproxy_client'
```
Comment (Abohessuin) [inline src/components/select_node_id.vue:53]: there is error here (Cannot find module 'tf_gridproxy_client' or its corresponding type declarations)

File: src/components/select_node_id.vue [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        country: country.value,
+        farmId: farm.value?.farmID,
+        farmName: farm.value?.name,
+        cru: props.deps.cpu,
+        mru: props.deps.memory ? Math.round(props.deps.memory / 1024) : undefined,
+        hru: props.deps.disk,
+        sru: props.deps.ssd,
+        publicIPs: props.deps.publicIp
+      })
+      .then((_nodes) => {
```
Comment (Abohessuin) [inline src/components/select_node_id.vue]: why using _nodes as kebab-case,i suggest make it camelCase as other variables also can we encapsulate 
the inside code of if(grid) into function with name that describe it is functionality and call it here

File: src/components/select_node_id.vue [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  async (publicIp) => {
+    farmLoading.value = true
+    const oldFarmId = farm.value
+    farm.value = undefined
+    farms.value = await getFarms(publicIp)
+    if (oldFarmId && farms.value.map((f) => f.farmID).includes(oldFarmId.farmID)) {
+      farm.value = oldFarmId
+    }
+    farmLoading.value = false
+  },
```
Comment (Abohessuin) [inline src/components/select_node_id.vue:93]: i suggest to encapsulate this into function and call it here

File: src/utils/get_farms.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    {
+      limit: totalCount,
+      where: {
+        deploymentData_contains: exclusiveFor,
+        state_eq: 'Created'
+      }
+    }
+  )
+
+  const nodeIds = new Set(nodeContracts.map((n) => n.nodeID))
```
Comment (Abohessuin) [inline src/utils/get_farms.ts:48]: it is better to name n to node  also it give me type error(parameter 'n' implicitly has an 'any' type) just check

File: src/utils/get_farms.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  const onlineFarms = await getOnlineFarms()
+
+  const blockedFarms = await getBlockedFarms(hasPublicIP, exclusiveFor)
+
+  return farms.filter((f) => onlineFarms.has(f.farmID) && !blockedFarms.has(f.farmID))
+}
+
+async function getOnlineFarms() {
+  const nodes = await gridProxyClient.nodes.listAll({ status: NodeStatus.Up })
+  return new Set(nodes.map((node) => node.farmId))
```
Comment (Abohessuin) [inline src/utils/get_farms.ts:21]: node give me error type Parameter 'node' implicitly has an 'any' type

**Review by Abohessuin — CHANGES_REQUESTED** [review]

File: src/components/select_node_id.vue (hunk 1 above)
Comment (MohamedElmdary) [reply src/components/select_node_id.vue:53]: I had to push `tf_gridproxy_client` new version to npm and should be fixed now

File: src/components/select_node_id.vue (hunk 3 above)
Comment (MohamedElmdary) [reply src/components/select_node_id.vue:93]: Shouldn't be a big deal as this logic won't be used anywhere else

File: src/utils/get_farms.ts (hunk 4 above)
Comment (MohamedElmdary) [reply src/utils/get_farms.ts:48]: I think this should be fixed after adding packages

File: src/utils/get_farms.ts (hunk 5 above)
Comment (MohamedElmdary) [reply src/utils/get_farms.ts:21]: This also should be fixed after adding packages

**Pull request merged**